### PR TITLE
Update key caching logic to incorporate custom storage

### PR DESCRIPTION
**Pull request by diptishiralkar-okta**
Resolves:OKTA-337701

### Steps to reproduce:

Actual result:

Expected result:

### CODE REVIEW

**Review by brennerryan-okta — APPROVED** [review]
LGTM

File: Sources/Keychain.swift [hunk 1]
```diff
@@ -38,16 +38,20 @@ public class OktaKeychain: NSObject {
...
 
-        let q = [
+        var q: [String: Any] = [
             kSecClass as String: kSecClassGenericPassword as String,
             kSecValueData as String: objectData!,
             kSecAttrAccount as String: key
-        ] as CFDictionary
+        ]
+
+        if #available(macOS 10.15, iOS 13.0, *) {
```
Comment (IldarAbdullin-okta) [inline Sources/Keychain.swift:47]: Can you change this macros for MacOS only. I would like to avoid potential regressions for existing iOS users

**Review by IldarAbdullin-okta — CHANGES_REQUESTED** [review]
Please remove iOS from supported platforms for the change

File: Sources/Keychain.swift (hunk 1 above)
Comment (diptishiralkar-okta) [reply Sources/Keychain.swift]: Done!

**Review by IldarAbdullin-okta — APPROVED** [review]
Nice! 👍

**Pull request merged**